src/index.ts

```diff
@@ -1,16 +1,37 @@
 import {
     AbstractTransactPlugin,
+    API,
+    Checksum256,
     TransactContext,
     TransactHookResponseType,
     TransactHookTypes,
+    TransactResult,
 } from '@wharfkit/session'
 
 /** Import JSON localization strings */
 import defaultTranslations from './translations.json'
 
-export class TransactPluginTemplate extends AbstractTransactPlugin {
+const DEFAULT_FINALITY_CHECK_DELAY = 150000 // 2.5 minutes
+
+interface TransactPluginFinalityCallbackOptions {
```

Comment:
I think we need to add a `logging` boolean to this as well, that would control whether or not the `log` method actually calls `console.log`. I don't think by default it should be using it, but since it exists, might be nice to let developers opt-in if they wanted to see status updates.